src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1291,10 +1293,28 @@ private string GetGetTypeInfoImplementation()
                     }
                 }
 
-                sb.Append(@"
+                sb.AppendLine(@"
     return null!;
 }");
 
+                // Explicit IJsonTypeInfoResolver implementation
+                string contextTypeName = _currentContext.ContextType.Name;
+
+                sb.AppendLine();
+                sb.Append(@$"{JsonTypeInfoTypeRef}? {JsonTypeInfoResolverTypeRef}.GetTypeInfo({TypeTypeRef} type, {JsonSerializerOptionsTypeRef} options)
+{{
+    {contextTypeName} context = this;
+
+    if (options != null && {OptionsInstanceVariableName} != options)
```

Comment:
should we throw ArgumentNullException on null instead?

Reply:
I figured it might be used to represent "just use the encapsulated options instance"